### PR TITLE
Preparations for Jitpack

### PR DESCRIPTION
**Pull request by jpooleycodes**
A couple of notes

- Cleaned up deprecated methods in gradle files
- Add maven publish to gdx-runtime and commons gradle (commons now has a version number as well)
- Support for downloading sources as well

Putting the following lines in my local runtime testing project in core project brings in both projects.

```
api "com.github.jpooleycodes.mundus:gdx-runtime:jitpack-SNAPSHOT"
```
and checking jitpack.io is showing downloads happening
image

Once these changes are merged I think we can finally create the 0.3.0 release

### CODE REVIEW

File: commons/build.gradle [hunk 1]
```diff
@@ -9,3 +13,24 @@ sourceSets.main.resources.srcDirs = ["src/main"]
...
 sourceSets.test.java.srcDirs = ["src/test"]
 sourceSets.test.resources.srcDirs = ["src/test"]
+
+task sourcesJar(type: Jar, dependsOn: classes) {
+    duplicatesStrategy(DuplicatesStrategy.EXCLUDE)
+    classifier = 'sources'
+    from sourceSets.main.allSource
+}
+
+publishing {
```
Comment (Dgzt) [inline commons/build.gradle:23]: I think with the root build.gradle modification this changes is unnecessary.

File: gdx-runtime/build.gradle [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
-def VERSION = '0.3.0'
 def NAME = 'mundus-gdx-runtime'
 
 task distRuntime(type: Jar, dependsOn: classes) {
-    baseName = NAME
-    version = VERSION
+    archivesBaseName = NAME
 
     dependsOn configurations.runtimeClasspath
     from {
```
Comment (Dgzt) [inline gdx-runtime/build.gradle:16]: Should we hold this gradle task if there is `publishToMavenLocal` task via maven-publish plugin.

Comment (jpooleycodes) [reply gdx-runtime/build.gradle:16]: We can get rid of but will need to update the working from sources wiki page which refers to them.

File: commons/build.gradle (hunk 1 above)
Comment (Dgzt) [reply commons/build.gradle:23]: No, this is needed.

**Review by Dgzt — APPROVED** [review]

**Pull request merged**